### ISSUE TITLE
Compilation error with GCC 4.8.4 nvcc 7.0.27

### ISSUE THREAD

**Issue by xanderdunn** [NVlabs/cub#36]
When I attempt to build cub on Ubuntu 14.04 with gcc 4.8.4 and nvcc 7.0.27, I get this error:

```
thread_load.cuh(423): error: reinterpret_cast cannot cast away const or other type qualifiers
          detected during:
            instantiation of "T cub::ThreadLoad(T *, cub::Int2Type<MODIFIER>, cub::Int2Type<1>) [with T=const pyramid::weight_type, MODIFIER=3]"
(442): here
            instantiation of "std::iterator_traits<InputIteratorT>::value_type cub::ThreadLoad<MODIFIER,InputIteratorT>(InputIteratorT) [with MODIFIER=cub::LOAD_CS, InputIteratorT=const pyramid::weight_type *]"
kernels.cu(56): here
```

It looks like [this](http://stackoverflow.com/questions/27995692/reinterpret-cast-casts-away-qualifiers) is related.

**Comment by xanderdunn** [comment]
It looks like this patch solves the issue: https://groups.google.com/forum/#!category-topic/cub-users/bugs-and-issues/TVdHx-jQYbU

**Comment by dumerrill** [comment]
Fixed in v1.5
